Test1/ConcurrentPriorityQueue/ConcurrentPriorityQueue.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace ConcurrentPriorityQueue;
+
+using System.Collections;
+
+/// <summary>Represents a thread-safe max priority queue.</summary>
+/// <typeparam name="TElement">Specifies the type of elements in the queue.</typeparam>
+/// <typeparam name="TPriority">Specifies the type of priority associated with enqueued elements.</typeparam>
+public class ConcurrentPriorityQueue<TElement, TPriority>
```

Comment:
```suggestion
public class ConcurrentPriorityQueue<TElement, TPriority> where TPriority : IComparable
```
Не всякий тип может быть приоритетом

Reply:
-0.5